azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/AzureAuthManager.java

```diff
@@ -66,7 +71,11 @@
         }
         return retrievers.retrieve().onErrorResumeNext(e ->
                 Single.error(new LoginFailureException(String.format("Cannot get credentials from authType '%s' due to error: %s", authType, e.getMessage())))
-        );
+        ).doOnSuccess(credentialWrapper -> {
+            if (StringUtils.isNotBlank(configuration.getHttpProxyHost())) {
```

Comment:
Do we need to set proxy before retrieve as we need to validate the token which may need network access

Reply:
seems we needn't to save proxy back to CredentialWrapper